### ISSUE TITLE
Lint/ToEnumArguments false negative for extra keyword arguments

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15257]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False negative

### Cop name

Lint/ToEnumArguments

### Steps to reproduce the problem

Create a file with this content:
```rb
def foo(x:)
  to_enum(__callee__, x: x, y: 1)
end

foo(x: 1).each {}
```

Run this command:
```sh
rubocop --only Lint/ToEnumArguments test.rb
```

RuboCop reports no offense.

At runtime, the enumerator raises:
```text
unknown keyword: :y (ArgumentError)
```

### Issue Reproduction:

<img width="536" height="178" alt="Image" src="https://github.com/user-attachments/assets/b87c84a7-8aac-4209-ae86-e456ae0160d3" />

### Expected behavior

`Lint/ToEnumArguments` should register an offense, because the enumerator re-invokes the current method with an extra keyword argument that the method does not accept.

### Actual behavior

No offense is registered.

### RuboCop configuration

Default configuration.

### RuboCop version

1.87.0

**Comment by RedZapdos123** [comment]
Opened a PR to resolve this issue.
